spec/unit/function_spec.rb

```diff
@@ -61,6 +61,22 @@
     end
   end
 
+  describe '#case' do
+    context 'when additional expression is provided' do
+      it 'returns an sql expression' do
+        expect(func.case({ 1 => "first" }, "last", users[:id]).sql_literal(ds))
+          .to eql(%((CASE "users"."id" WHEN 1 THEN 'first' ELSE 'last' END)))
+      end
+    end
+
+    context 'when additional expression is not provided' do
+      it 'returns an sql expression' do
+        expect(func.case({ 1 => "first" }, "last").sql_literal(ds))
+          .to eql(%((CASE WHEN 1 THEN 'first' ELSE 'last' END)))
```

Comment:
This SQL is not valid, at least not in PostgreSQL/Oracle, it's better to have a working example, we don't want to confuse people :)

Reply:
Right, good point! I thought about literal value for `true`, but I'd forgotten that MySQL and PostgreSQL have a different mapping for Integer and string in that case.